### DISCUSSION TITLE
vanX.reactive assignment via onclick not working

### DISCUSSION THREAD

**Discussion by richcorbs** [vanjs-org/van#201] [Q&A] [answered]
I just found VanJS today and I'm loving it so far! I really like how light and simple and powerful it is!

However, I can't get this to work. The onclick doesn't seem to set currentContent. Any ideas?

```
let contents = vanX.reactive([{ Live: true, Title: 'Title', Slug: 'slug', Text: 'text', UpdatedAt: '2022-01-01' }])
let currentContent = vanX.reactive({})
const Hello = () => {
  return
    div(
      div(
        contents.map(content =>
          div(
            { onclick: () => currentContent = content },
            content.Title
          )
        )
      ),
      div(
        input({ type: 'text', value: () => currentContent.Title, oninput: e => currentContent.Title = e.target.value })
      )
    )
}

**Comment by richcorbs** [comment]
Do I somehow need vanX.list?

**Comment by Tao-VanJS** [comment, marked as answer]
To make it work, you can't re-assignment the variable `currentContent`. Instead, you can set the sub-fields of `currentContent`, something like:

```js
currentContent.Title = ...
```

**Comment by richcorbs-ncino** [comment]
Inside of the onclick I can set the individual keys. Is there a way I could call out to a function from the onclick that would do this for me? Like `onclick: () => setContent(content)`?

**Reply by richcorbs-ncino** [reply]
I have this function in place and it console.logs the right things but it doesn't seem to be reactive in the end.

**Reply by richcorbs-ncino** [reply]
I tried to delete existing keys from currentContent that might get in the way and did it wrong. This was my mistake.

**Reply by Tao-VanJS** [reply]
Hi @richcorbs,

> I tried to delete existing keys from currentContent that might get in the way and did it wrong. This was my mistake.

To confirm, is your problem resolved now?

**Reply by richcorbs** [reply]
Yes, thank you.
